src/openforms/authentication/contrib/org_oidc/plugin.py

```diff
@@ -23,20 +41,25 @@ class OIDCAuthentication(BasePlugin):
     provides_auth = AuthAttribute.employee_id
 
     def start_login(self, request: HttpRequest, form: Form, form_url: str):
-        login_url = reverse("org-oidc:init", request=request)
-
-        auth_return_url = reverse(
+        auth_return_url = _reverse_plus(
             "authentication:return",
             kwargs={"slug": form.slug, "plugin_id": self.identifier},
-        )
-        return_url = furl(auth_return_url).set(
-            {
+            request=request,
+            query={
                 "next": form_url,
-            }
+            },
         )
-
-        redirect_url = furl(login_url).set({"next": str(return_url)})
-        return HttpResponseRedirect(str(redirect_url))
+        redirect_url = _reverse_plus(
+            "org-oidc:init",
+            request=request,
+            query={
+                "next": auth_return_url,
+            },
+        )
+        if request.user.is_authenticated:
+            # logout user if logged in with other account
+            auth.logout(request)
```

Comment:
can you expand this comment a bit that we do this because otherwise staff users get redirect to internal admin pages instead of the intended next page due to some internal oidc-library state in the session?